core/core.mk

```diff
@@ -11,6 +11,17 @@ include mk/config.mk
 # $(ARCH).mk also sets the compiler for the core module
 include core/arch/$(ARCH)/$(ARCH).mk
 
+ifeq ($(CFG_OS_REV_REPORTS_GIT_SHA1),y)
+ifeq ($(arch-bits-core),64)
+git-sha1-len := 16
+else
+git-sha1-len := 8
+endif
+TEE_IMPL_GIT_SHA1 := 0x$(shell git rev-parse --short=$(git-sha1-len) HEAD 2>/dev/null | cut -c -$(git-sha1-len) || echo 0)
```

Comment:
Since we pipe through `cut -c -$(git-sha1-len)`, do we need the `--short=$(git-sha1-len)` parameter?

Reply:
No we don't. Updated.

Reply:
--short=$(git-sha1-len) will also change the number of character printed for 64bits platforms.
If nothing is specified, the number of characters will depend on each person's git configuration. Looking at the man page, it will default to at least 4. 
If the 4 first characters aren't ambiguous, it will remain 4 AFAICT. Else, I guess the command will add one more until it's not.

Reply:
> --short=$(git-sha1-len) will also change the number of character printed for 64bits platforms.

Yes and I believe that's fine.

> If nothing is specified, the number of characters will depend on each person's git configuration. [...]

Right. So `--short=$(git-sha1-len)` is needed after all. Updated, thanks!